### ISSUE TITLE
Tab Groups don't match any open window...

### ISSUE THREAD

**Issue by eguerin** [Drive4ik/simple-tab-groups#99]
Doesn't seem like the Tab Groups are picking up the active windows, I have 15 tabs open, but none of the Tab groups are representative of the active window.   If I go an manage the tab groups, I see old version of my windows, but nothing in the current window setup.

Is there something I need to do in order for it to work like old Tab Groups

**Comment by meru448** [comment]
Hey there, experiencing a similar or the same issue w/ the "phantom" tab group - so if you try to load the phantom tab group, it will only load the tabs that are shown in it, even though that is your "active" group & you will lose any tabs from the moment that tab changed to a "phantom" group, what causes this I'm not sure, as the group remained functional (showing increasing & decreasing tabs) through 2-3 restarts of firefox, but then became a "phantom" group at a later date.

As to how to fix it: make a new group, and move your tabs into that group then delete the phantom group. this new group should remain a non-phantom group for a time, atleast until you restart firefox.

**Comment by eguerin** [comment]
This is seriously frustrating, it basically makes this Add On completely useless, doesn't remember any current windows configuration, which makes this just a glorified bookmarks manager.

**Comment by eguerin** [comment]
Unfortunately your suggestion doesn't work because half my open tabs are not listed or available in any of the Groups when I go to manage the tab groups visually.   There also doesn't appear to be a way to get them from a tab to a group manually.

**Comment by misterHippo** [comment]
G'day all, 

I've been experiencing / trying to ignore the same issue. Upon launch of Firefox, my current active tab group appears to load, but it isn't actually selected in the S.T.G. system. Therefore, when I change tab groups, the items I presume are assigned to one need to be pushed into a group - or be lost.

I've almost trained myself to just choose a group upon launch, but I see this here - so I thought I'd stop by and share my info.

I remember something a few months back, where I was getting new groups I hadn't created. I wonder if this issue is tied to that issue (or its resolution)?

Keep up the good work @Drive4ik  :)

**Comment by Drive4ik** [comment]
Hi all.
Can you help me understand this problem?
First of all create backup all your data:
![screenshot_3](https://user-images.githubusercontent.com/7843031/37601671-55f75334-2b93-11e8-9764-a48659beea5a.png)

then disable STG, download and install this TEST build of STG:
rename file from
```
test_startup_simple_tab_groups-2.4.8-an+fx.xpi.zip
to
test_startup_simple_tab_groups-2.4.8-an+fx.xpi
```

[test_startup_simple_tab_groups-2.4.8-an+fx.xpi.zip](https://github.com/Drive4ik/simple-tab-groups/files/1825381/test_startup_simple_tab_groups-2.4.8-an.fx.xpi.zip)

restore backup, and try to replicate this bug.

if you replicate this bug you can see this notification
![screenshot_1](https://user-images.githubusercontent.com/7843031/37602056-3d1311e0-2b94-11e8-85b0-7f80faae2b41.png)

you need at once go to addon settings, scroll down and press this button
![screenshot_2](https://user-images.githubusercontent.com/7843031/37602098-5583c3b4-2b94-11e8-9b9a-137d1bf8d5fa.png)

send to me this file by email: drive4ik@gmail.com

P.S. all of your private data in this logs are replaced by security and your urls replaced by URL_1, URL_2, URL_3, URL_1 etc... You can open this file and see it.

Thanks

**Comment by Drive4ik** [comment]
Hi! Can you retest this bug in latest version? (v2.5)
If bug was fixed - close issue.

Thanks

**Comment by misterHippo** [comment]
Sorry I didn't get a chance to do what you requested above. I've restarted Firefox, and now there is a little loading gif that is going around in circles on the menu bar preventing me from opening the group list.

Please tell me my tabs aren't lost forever?

**Comment by misterHippo** [comment]
Woooo. Ok, for anyone else suffering as I was, go here: https://addons.mozilla.org/en-US/firefox/addon/simple-tab-groups/versions/

You can get the previous version to version 2.5. That restores functionality.

**Comment by misterHippo** [comment]
Ok, I have lead for you.

When updating to v2.5 for the second time, I saw an error.
![image](https://user-images.githubusercontent.com/605912/37744789-331637d4-2dc5-11e8-8de5-cd7505125479.png)

Now, when I updated the first time, I noted that I didn't have a group selected. (as I had launched Firefox, and updated.) While I was relaunching with v2.5 I wondered if I should have pre-selected a group. Anyway, I've since tested updating after selecting a group first. And the above message still appears. So perhaps it's not that simple?

But maybe Group 1 should be the default if none are pre-selected.

**Comment by Drive4ik** [comment]
@misterHippo Hi. This download can last no more than 30 seconds. so it means that there are tabs due to which synchronization can go wrong

**Comment by misterHippo** [comment]
so having 85 tabs open may be causing difficulties?

**Comment by Drive4ik** [comment]
No, I meant that if some tabs are loaded then the program is waiting for them. because while they are being loaded I do not know their url ... so I have to wait for the correct synchronization.
I fix bug in notification tomorrow.
Sorry.

**Comment by misterHippo** [comment]
Peace. It's all good. I was quietly freaking out. ... sort of quietly. ;) Rolling back ceased the freak-out. :)

**Comment by Drive4ik** [comment]
@misterHippo What FF version you are used?

**Comment by Drive4ik** [comment]
@misterHippo Do you have any problems with latest v2.5.1?

**Comment by eguerin** [comment]
I will retry this on the new version when I get some time.

**Comment by Drive4ik** [comment]
@eguerin Thanks.

**Comment by misterHippo** [comment]
G'day @Drive4ik :)

My Firefox version: 59.0.1 (64 bit) 

I will try the new STG version (2.5.1) and edit this comment.

**Comment by eguerin** [comment]
The tabs I have open now still don't show in any group.    I have to run through your test process though which will mean moving all the tabs into a group and then doing a backup and using your debug version, I will do that and report back.

**Comment by eguerin** [comment]
I think part of the issue is I also have multiple Firefox windows open, and maybe it doesn't know where the windows are?

**Comment by Drive4ik** [comment]
@eguerin 

> The tabs I have open now still don't show in any group.

You can manualy sync group and window, for example
Group 1 - window 1
Group 2 - window 2
Group 3 - window 3
and then "normal" restart FF (close-wait-open)
After restart you should get the same state:
Group 1 - window 1
Group 2 - window 2
Group 3 - window 3

If it's not happend, please tell me with all the details what happen after restart :)

**Comment by Drive4ik** [comment]
Please, do this with latest stable (not test) version.
Thanks.
